execution/executor/src/speculation_cache/mod.rs

```diff
@@ -128,9 +128,10 @@ impl SpeculationCache {
     }
 
     pub fn new_with_tree_state(tree_state: TreeState) -> Self {
+        let executor_trees = ExecutedTrees::from(tree_state);
         Self {
-            synced_trees: ExecutedTrees::new_empty(),
-            committed_trees: ExecutedTrees::from(tree_state),
+            synced_trees: executor_trees.clone(),
+            committed_trees: executor_trees,
```

Comment:
I'm a bit confused. For the old code, it seems that it's possible to initialize `synced_trees` and `committed_trees` to two different trees, while the new code always initializes them as the same thing. Comment a bit?

Reply:
The other constructor used for normal operation still has differenet synced & committed. This is for the bootstrapper which doesn't distinguish the two states.

The change here is to change synced_trees from empty to the same with commited_trees, which otherwise will fail the check of `first_version_to_keep <= num_persistent_txns` in `commit_blocks()`.